deploy/overlays/namespace_scoped/kustomization.yaml

```diff
@@ -0,0 +1,19 @@
+bases:
+  - ../../base
+
+resources:
+  - rbac.yaml
+
+replacements:
+  - source:
+      kind: Deployment
+      fieldPath: metadata.namespace
+    targets:
+      - select:
+          kind: Deployment
+        fieldPaths:
+          - spec.template.spec.containers.[name=manager].args.3
```

Comment:
Feels rather risky to write it like this since if we change anything in base it will the order of the args can get changed.
I would probably just patch the entire args fields to be sure.

Reply:
Yes, this is very brittle but I couldn't find a better way to set the value of a container arg in an overlay. Do you know if there is a better way? The goal is to:

- set `--namespace=<namespace>` in namespaced mode
- set omit `--namespace` or set to empty string in cluster scoped mode